### PR TITLE
[Merged by Bors] - feat: adapt tactics to use `optConfig` instead of `(config)?`

### PR DESCRIPTION
**Pull request by kmill**
Except for `convert` and `convert_to`, tactics now use the new `optConfig` configuration option syntax.

The problem with `convert` and `convert_to` is that `optConfig` isn't compatible with a term following it. This should eventually be possible, but it will depend on core changes.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Measurability.lean [hunk 1]
```diff
@@ -29,7 +29,7 @@ macro "measurability" : attr =>
 The tactic `measurability` solves goals of the form `Measurable f`, `AEMeasurable f`,
 `StronglyMeasurable f`, `AEStronglyMeasurable f μ`, or `MeasurableSet s` by applying lemmas tagged
 with the `measurability` user attribute. -/
-macro "measurability" (config)? : tactic =>
+macro "measurability" _cfg:optConfig : tactic =>
```
Comment (kim-em) [inline Mathlib/Tactic/Measurability.lean:32]: Why the `_cfg` here?

**Comment by kim-em** [comment]
bors d+

File: Mathlib/Tactic/Measurability.lean (hunk 1 above)
Comment (kmill) [reply Mathlib/Tactic/Measurability.lean:32]: These tactics apparently don't use their configurations. The reason the syntax worked before is that `macro` will turn the parser `(config)?` into something like `_:(config)?`

Comment (kmill) [reply Mathlib/Tactic/Measurability.lean:32]: I think I'll delete the configuration actually. I was going for doing one-to-one changes with this PR, but having an unused configuration is bound to be confusing to someone.

**Pull request closed without merging**